src/Haskell/Ide/Engine/Transport/LspStdio.hs

```diff
@@ -795,6 +795,26 @@ reactor inp diagIn = do
 
 -- ---------------------------------------------------------------------
 
+getFormattingProvider :: R FormattingProvider
+getFormattingProvider = do
+  providers <- asks formattingProviders
+  lf <- asks lspFuncs
+  mc <- liftIO $ Core.config lf
+  -- LL: Is this overengineered? Do we need a pluginFormattingProvider
+  -- or should we just call plugins straight from here based on the providerType?
```

Comment:
I would personally not have this complication and just call the plugin directly.

Reply:
I think having an entry in the plugin is a good thing.

Eventually we can manage config based on the installed providers too, and it opens the way to dynamically (or statically but from config) loading plugins.  Which in my opinion has always been a goal.